fe/fe-core/src/main/java/org/apache/doris/common/util/HiveViewSqlTransformer.java

```diff
@@ -0,0 +1,66 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.common.util;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class for transforming Hive view SQL to ensure case consistency.
+ * This transformer converts SQL keywords, table names, and column names to lowercase
+ * while preserving the case of string literals enclosed in quotes.
+ */
+public class HiveViewSqlTransformer {
```

Comment:
convert SQL without parser and analyzer is not safe.
Recently I refactor the whole case sensibility issue: https://github.com/apache/doris/pull/52561
And for hive catalog, you can try adding `"only_test_lower_case_table_names" = "2"` to see if it can solve your problem